### PR TITLE
Fix Create app selection updating when entities are deleted

### PR DESCRIPTION
**Pull request by ctrlaltdavid**

### CODE REVIEW

**Comment by ctrlaltdavid** [comment]
### Test Plan

1) Test single entity deletion, entities list:
- Run the create30SecondEntity.js script to create a sphere with a lifetime of 30s.
- Open the Create app. Click on the entity just created to select it. The edit handles etc. should appear as usual. And the entity should be highlighted in the entity list.
- Wait for the entity to be automatically deleted once its lifetime has expired.
- The edit handles etc. should disappear.
- The entity's entry in the entity list should unhighlight and disappear.
- The count of the number of entities selected at the bottom of the entity list should change from 1 selected to N found.

2) Test single entity deletion, properties tab:
- Per 1) except display the properties tab.
- When the entity is automatically deleted, the properties tab should change to display "No selection" (top left of dialog).

3) Test multiple entities deletion, entities list:
- Per 1) except run the create30SecondEntities.js script, which creates 3 spheres. Select 2 of these spheres at the selection step.

4) Test multiple entities deletion, properties tab:
- Per 2) except run the create30SecondEntities.js script, which creates 3 spheres. Select 2 of these spheres at the selection step.

5) Test other entity deletion, single entity selected, entities list:
- Create a box entity.
- Run the create30SecondEntity.js script to create a sphere with a lifetime of 30s.
- Open the Create app. Click on the box entity to select it. The edit handles etc. should appear as usual.
- Open the entities list. The entity should be highlighted in the entity list.
- Wait for the sphere entity to be automatically deleted once its lifetime has expored.
- The selected box entity should remain selected: edit handles should continue to display and the entity should continue to be highlighted in the entity list.

6) Test other entity deletion, single entity selected, properties tab:
- Per 5) except open the properties tab instead of the entities list. When the entity is automatically deleted, the content of the properties tab should not change.

7) Test deletion of entity in multiple selection, entities list:
- Create another box entity.
- Run the create30SecondEntity.js script to create a sphere with a lifetime of 30s.
- Open the Create app. Click on both box entities and the sphere entity to select them all. The edit handles etc. should appear as usual.
- Open the entities list. The entries for the 3 entities selected should be highlighted.
- Wait for the sphere entity to be automatically deleted once its lifetime has expired.
- The two box entities should remain selected: edit handles should continue to display; the entity list should change to have just the two entities highlighted.

8) Test deletion of entity in multiple selection, properties tab:
- Per 7) except open the properties tab instead of the entities list. It should initially say "Multiple 3"  at the top left. When the entity is automatically deleted, the properties tab should change to say "Box (2)" at the top left.

[21893.zip](https://github.com/highfidelity/hifi/files/2108502/21893.zip)

https://highfidelity.manuscript.com/f/cases/15910

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10647/HighFidelity-Beta-PR13397-e8154b5.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10647/HighFidelity-Beta-PR13397-e8154b5.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10647/HighFidelity-Beta-Interface-PR13397-e8154b5.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10647/HighFidelity-Beta-PR13397-e8154b5.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10647/HighFidelity-Beta-Interface-PR13397-e8154b5.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Fail @conklin94122 
While on the List tab, the entity will unhighlight but still remain in the list and the total entities number at the bottom of the list does not update with new number of entities. Going to Properties and coming back to list tab does update it.

Logs and Video attached:
[PR13397 Files.zip](https://github.com/highfidelity/hifi/files/2111916/PR13397.Files.zip)

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10704/HighFidelity-Beta-PR13397-064f4ca.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10704/HighFidelity-Beta-PR13397-064f4ca.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10704/HighFidelity-Beta-Interface-PR13397-064f4ca.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10704/HighFidelity-Beta-PR13397-064f4ca.exe

File: scripts/system/html/js/entityList.js [hunk 1]
```diff
@@ -195,6 +195,18 @@ function loaded() {
           }
       }
 
+      function removeEntities(deletedIDs, selectedIDs) {
+          for (i = 0, length = deletedIDs.length; i < length; i++) {
```
Comment (dback2) [inline scripts/system/html/js/entityList.js]: This code matches the delete code at lines 385-388, can we make them share this?

File: scripts/system/edit.js [hunk 2]
```diff
@@ -475,15 +475,16 @@ var toolBar = (function () {
...
-        // Allow for multiple entity deletes before updating the entity list.
+        // Allow for multiple entity deletes before updating the entities selected.
         entitiesToDelete.push(entityID);
         if (deletedEntityTimer !== null) {
             Script.clearTimeout(deletedEntityTimer);
         }
         deletedEntityTimer = Script.setTimeout(function () {
-            selectionManager.removeEntities(entitiesToDelete);
-            entityListTool.clearEntityList();
-            entityListTool.sendUpdate();
```
Comment (dback2) [inline scripts/system/edit.js]: Do we not need a sendUpdate anymore because we are depending on the refresh list timer to update?

Comment (dback2) [reply scripts/system/edit.js]: Oh wait my bad that doesn't trigger an update in the end

File: scripts/system/html/js/entityList.js [hunk 3]
```diff
@@ -367,6 +379,8 @@ function loaded() {
                       updateSelectedEntities(data.selectedIDs);
                       resize();
                   }
+              } else if (data.type === "removeEntities" && data.deletedIDs !== undefined) {
```
Comment (dback2) [inline scripts/system/html/js/entityList.js]: If selectedIDs needs to be not undefined for update above, it doesn't need to be the same here?

File: scripts/system/html/js/entityProperties.js [hunk 4]
```diff
@@ -893,9 +893,9 @@ function loaded() {
                     } else {
                         elServerScriptStatus.innerText = "Not running";
                     }
-                } else if (data.type === "update") {
```
Comment (dback2) [inline scripts/system/html/js/entityProperties.js]: Redundant check for data.selections on line 914

Comment (ctrlaltdavid) [reply scripts/system/html/js/entityProperties.js]: Removed redundant check.

File: scripts/system/html/js/entityList.js (hunk 3 above)
Comment (ctrlaltdavid) [reply scripts/system/html/js/entityList.js]: Added extra validation check.

File: scripts/system/html/js/entityList.js [hunk 5]
```diff
@@ -195,6 +195,18 @@ function loaded() {
...
 
+      function removeEntities(deletedIDs, selectedIDs) {
+          for (i = 0, length = deletedIDs.length; i < length; i++) {
+              delete entities[deletedIDs[i]];
+              entityList.remove("id", deletedIDs[i]);
+          }
+          if (refreshEntityListTimer) {
+              clearTimeout(refreshEntityListTimer);
+          }
+          refreshEntityListTimer = setTimeout(refreshEntityListObject, 50);
```
[the remark this answers is not in the document]
Comment (ctrlaltdavid) [reply scripts/system/html/js/entityList.js]: Refactored.

File: scripts/system/html/js/entityList.js (hunk 1 above)
Comment (ctrlaltdavid) [reply scripts/system/html/js/entityList.js]: Refactored.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10786/HighFidelity-Beta-PR13397-d43be5c.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10786/HighFidelity-Beta-PR13397-d43be5c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10786/HighFidelity-Beta-Interface-PR13397-d43be5c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10786/HighFidelity-Beta-PR13397-d43be5c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10786/HighFidelity-Beta-Interface-PR13397-d43be5c.exe

**Comment by Qliemillar** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10902/HighFidelity-Beta-PR13397-d43be5c.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10902/HighFidelity-Beta-PR13397-d43be5c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10902/HighFidelity-Beta-Interface-PR13397-d43be5c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10902/HighFidelity-Beta-PR13397-d43be5c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10902/HighFidelity-Beta-Interface-PR13397-d43be5c.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
List updates successfully after deletion, other test steps pass as well.

**Pull request merged**